### PR TITLE
fix: ga4 issues

### PR DESCRIPTION
**Pull request by mihir-4116**
## PR Description

- removed anonymousId mapping with userId
- for hybrid mode we removed clientId since it's going to map with anonymousId
- sendUserId flag is also removed, whenever we have userId we will send it to ga4

## Notion ticket

Ticket link

## Screenshots

Please add screenshots for any new features or UI bug fixes for the following browsers -

- Chrome
  >
- Firefox
  >
- Safari
  >

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: src/integrations/GA4/browser.js [hunk 1]
```diff
@@ -83,8 +73,7 @@ export default class GA4 {
 
   init() {
     // To do :: check how custom dimension and metrics is used
-    const userId = this.analytics.userId || this.analytics.anonymousId;
-    this.loadScript(this.measurementId, userId);
+    this.loadScript(this.measurementId);
```
Comment (shrouti1507) [inline src/integrations/GA4/browser.js:76]: @mihir-4116 can you please add comment here regarding why we are removing userId from loading script?

File: src/integrations/GA4/browser.js [hunk 2]
```diff
@@ -165,6 +154,9 @@ export default class GA4 {
     if (this.addSendToParameter) {
       params.send_to = this.measurementId;
     }
+    if (this.analytics.userId) {
+      params.user_id = this.analytics.userId;
```
Comment (shrouti1507) [inline src/integrations/GA4/browser.js:158]: same here, please add a comment that says, what is achieved by adding user_id in params

File: src/integrations/GA4/browser.js [hunk 3]
```diff
@@ -240,20 +232,26 @@ export default class GA4 {
...
+
+    if (this.capturePageView === 'rs') {
+      let pageProps = rudderElement.message.properties;
+      if (!pageProps) return;
+      pageProps = flattenJsonPayload(pageProps);
+      const properties = { ...getPageViewProperty(pageProps) };
+      if (this.addSendToParameter) {
+        properties.send_to = this.measurementId;
+      }
+      if (this.analytics.userId) {
```
Comment (shrouti1507) [inline src/integrations/GA4/browser.js:244]: same here

File: src/integrations/GA4/browser.js [hunk 4]
```diff
@@ -224,7 +216,7 @@ export default class GA4 {
     logger.debug('In GoogleAnalyticsManager Identify');
     window.gtag('set', 'user_properties', flattenJsonPayload(this.analytics.userTraits));
     if (this.sendUserId && rudderElement.message.userId) {
-      const userId = this.analytics.userId || this.analytics.anonymousId;
+      const { userId } = rudderElement.message;
```
[the remark this answers is not in the document]
Comment (mihir-4116) [reply src/integrations/GA4/browser.js:220]: updated

File: src/integrations/GA4/browser.js (hunk 2 above)
Comment (mihir-4116) [reply src/integrations/GA4/browser.js:158]: updated

File: src/integrations/GA4/browser.js (hunk 3 above)
Comment (mihir-4116) [reply src/integrations/GA4/browser.js:245]: updated

File: src/integrations/GA4/browser.js (hunk 1 above)
Comment (mihir-4116) [reply src/integrations/GA4/browser.js:76]: added comment in loadScript and identify function

**Review by MoumitaM — APPROVED** [review]

**Comment by mihir-4116** [comment]
raised separate prs for all the issues and merged them

**Pull request closed without merging**